api/migrations/0005_update_profile.py

```diff
@@ -0,0 +1,57 @@
+# Generated by Django 3.1.4 on 2021-02-15 19:12
+
+import django.contrib.auth.models
+import django.db.models.deletion
+from django.conf import settings
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+        ("api", "0004_plural_match_status"),
+    ]
+
+    operations = [
+        migrations.AlterModelOptions(
+            name="profile",
+            options={"verbose_name": "Profile", "verbose_name_plural": "Profiles"},
+        ),
+        migrations.AlterModelManagers(
+            name="profile",
+            managers=[
+                ("objects", django.contrib.auth.models.UserManager()),
+            ],
+        ),
+        migrations.RemoveField(
+            model_name="profile",
+            name="first_name",
+        ),
+        migrations.RemoveField(
+            model_name="profile",
+            name="id",
+        ),
+        migrations.RemoveField(
+            model_name="profile",
+            name="last_name",
+        ),
+        migrations.RemoveField(
+            model_name="profile",
+            name="user",
+        ),
```

Comment:
Explain this to me like I'm in first grade: Why is this being removed and what happens to these fields?

Reply:
Because this is a generated file, it's usually easier to talk about the changes on the actual model in models.py which is what ultimately creates this file.

Anyway, these fields in the profile are redundant because they already exist within the user model. By extending the user model, we get the functionality we want for free. The id and user fields are unneeded because the implicit user_ptr field can double as a primary key and a reference to the related user. The first and last name exist in the user model and don't need to be repeated in the profile.

Reply:
Thanks!